pkg/errors/error.go

```diff
@@ -50,22 +50,35 @@ const (
 	errDatabase     = errClass("database")
 )
 
+func isCodeCancellation(code gcodes.Code) bool {
+	return code == gcodes.Canceled || code == gcodes.DeadlineExceeded
+}
+
+func isErrorCancellation(err error) bool {
+	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+		return true
+	}
+
+	if st, ok := status.FromError(err); ok {
+		return isCodeCancellation(st.Code())
+	}
+
+	return false
+}
+
 // classifyError classifies given error by type and internals. Successfully classified (not errNoClass) errors don't get reported to sentry.
 func classifyError(err error) errClass {
+	if isErrorCancellation(err) {
+		return errCancellation
+	}
+
 	if st, ok := status.FromError(err); ok {
-		if st.Code() == gcodes.Canceled {
```

Comment:
Removed this to group related logic together (cost is re-running `status.FromError`)